### DISCUSSION TITLE
Not able to get the scanned data for Zebra barcode scanners

### DISCUSSION THREAD

**Discussion by aditya-babu1** [capacitor-community/bluetooth-le#587] [Q&A]
**Describe the bug**
We are able to connect with the Zebra Barcode device(DS2278 model), but after scanning data transmission is not happening, does this plugin support all the devices in low energy mode?

**Plugin version:**
- @capacitor-community/bluetooth-le: 2.2.3

**Comment by pwespi** [comment]
Yes, this plugin should support all Bluetooth Low Energy devices (however not Bluetooth classic).

Do you know the Bluetooth Low Energy protocol used by this Zebra device? You will have to use `startNotifications` with a specific service and characteristic to obtain any data.

**Comment by aditya-babu1** [comment]
yes The Zebra device which we are using has the appropriate low-energy barcodes(Cradle Bluetooth Low Energy, HID Bluetooth Low Energy, SSI Bluetooth LE) for using in Bluetooth Low Energy mode => Tried all of these
yes @pwespi, we tried using startNotifications with all service and characteristic ids having notify/indicate true individually, also tried enabling notifications using appropriate descriptor id and then used startNotifications with appropriate service and characteristic id, but data-transfer is not happening from zebra barcode scanner, do you see anything wrong here, can you please help on this?

**Comment by pwespi** [comment]
Do you have the specification of the Zebra device? Maybe you need to write some configuration to a characteristic to enable the data transmission.

**Comment by aditya-babu1** [comment]
@pwespi, No I am not able to find the developer manual for DS2278 model, but found a developer manual for zebra printers, there it's suggesting to write some configuration to a characteristic to enable the data transmission.
do you know if there is any standard way to get the appropriate service and characteristic ids for enabling data transmission?
Developer manual for zebra printers
https://www.zebra.com/content/dam/zebra/software/en/application-notes/AppNote-BlueToothLE-v4.pdf

**Reply by pwespi** [reply]
No, there is no standard way to enable data transmission. This is usually a custom implementation of the manufacturer of such devices.

If you don't have access to the specification of your peripheral, you can try to reverse engineer it using an app like "nRF connect for mobile" and play around with the various services and characteristics.
